pkg/manager/member/pd_scaler.go

```diff
@@ -31,26 +32,32 @@ import (
 
 type pdScaler struct {
 	generalScaler
+	pdControl pdapi.PDControlInterface
 }
 
 // NewPDScaler returns a Scaler
 func NewPDScaler(pdControl pdapi.PDControlInterface,
 	pvcLister corelisters.PersistentVolumeClaimLister,
 	pvcControl controller.PVCControlInterface) Scaler {
-	return &pdScaler{generalScaler{pdControl, pvcLister, pvcControl}}
+	return &pdScaler{generalScaler{pvcLister, pvcControl}, pdControl}
 }
 
-func (psd *pdScaler) Scale(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+func (psd *pdScaler) Scale(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
 	scaling, _, _, _ := scaleOne(oldSet, newSet)
 	if scaling > 0 {
-		return psd.ScaleOut(tc, oldSet, newSet)
+		return psd.ScaleOut(meta, oldSet, newSet)
 	} else if scaling < 0 {
-		return psd.ScaleIn(tc, oldSet, newSet)
+		return psd.ScaleIn(meta, oldSet, newSet)
 	}
-	return psd.SyncAutoScalerAnn(tc, oldSet)
+	return psd.SyncAutoScalerAnn(meta, oldSet)
 }
 
-func (psd *pdScaler) ScaleOut(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+func (psd *pdScaler) ScaleOut(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	tc, ok := meta.(*v1alpha1.TidbCluster)
```

Comment:
Can we get the full TC info from the meta here?

Reply:
After the manual test, the answer is yes.